File: Algorithms/Searching/BinarySearch/BinarySearchIterative.py
from typing import List

class BinarySearchIterative:
    def binary_search_iterative(self, arr: List[int], target: int) -> int:
        left, right = 0, len(arr) - 1

        while left <= right: # Loop until pointers meet
            mid = left + (right - left) // 2

            # Check if the mid point is target
            if arr[mid] == target:
                return mid   # Target found, return index
            elif arr[mid] < target:
                left = mid + 1 # Adjust the left pointer and move towards right
            else:
                right = mid - 1   # Adjust the right pointer and move towards left
            
        return -1
    
def main():
    sol = BinarySearchIterative()
    arr = [3, 6, 8, 12, 14, 19, 24, 30, 35, 42]
    target = 19
    res = sol.binary_search_iterative(arr, target)
    print(res)

if __name__ == "__main__":
    main()

File: Algorithms/Searching/BinarySearch/JavaCode/BinarySearchIterative.java
package Algorithms.Searching.BinarySearch;

public class BinarySearchIterative {
    static int binarySearch(int[] arr, int x) {
        int l = 0;
        int r = arr.length - 1;

        while(l <= r) {
            int mid = l + (r - l) / 2;

            if (arr[mid] == x) {
                return mid; 
            }

            if (x > arr[mid]) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        int arr[] = {2,5,7,8,11,19};
        int x = 11;

        int res = binarySearch(arr, x);
        if (res == -1)
        {
            System.out.println("the element is not present in the array");
        } else {
            System.out.println("The element is present at index " + res);
        }
    }
}